againHelpers.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+#Copyright (C) 2013 Niklas Thorne.
+#
+#This file is part of again.
+#
+#again is free software: you can redistribute it and/or modify
+#it under the terms of the GNU General Public License as published by
+#the Free Software Foundation, either version 3 of the License, or
+#(at your option) any later version.
+#
+#again is distributed in the hope that it will be useful,
+#but WITHOUT ANY WARRANTY; without even the implied warranty of
+#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#GNU General Public License for more details.
+#
+#You should have received a copy of the GNU General Public License
+#along with again.  If not, see <http://www.gnu.org/licenses/>.
+
+readonly DATE_FORMAT="%Y-%m-%d"
+
+# Adjust $ORIGINAL by $ADJUST, store in $NEW_DATE
+function adjust_date()
+{
+  ADJUST_NUM=`expr match "$ADJUST" '+\?\([0-9]\+\)'`
```

Comment:
expr match isn't supported on BSD according to the man page ("these arguments are treated just as their respective string values", FreeBSD 10.2).